### PR TITLE
Fix configuration for kotlinx.train

### PR DESCRIPTION
**Pull request by osipxd**
**Subsystem**
Infrastructure

**Motivation**
Isolate kotlinx.train configuration better and fix versions substitution

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -142,7 +97,7 @@ subprojects {
     configureCodestyle()
 }
 
-println("Using Kotlin compiler version: ${org.jetbrains.kotlin.config.KotlinCompilerVersion.VERSION}")
+println("Using Kotlin compiler version: ${libs.versions.kotlin.get()}")
```
[the remark this answers is not in the document]
Comment (osipxd) [reply build.gradle.kts:100]: Yes, as we override the kotlin version alias with the value from `kotlin_version` in `settings.gradle.kts`

File: buildSrc/src/main/kotlin/Train.kt [hunk 2]
```diff
@@ -46,35 +47,26 @@ fun Project.filterSnapshotTests() {
...
-    val skipSnapshotChecks = rootProject.properties["skip_snapshot_checks"] != null
-    if (!skipSnapshotChecks) {
-        check(version, atomicfu_version, "atomicfu")
-        check(version, coroutines_version, "coroutines")
-        check(version, serialization_version, "serialization")
-    }
-    repositories {
-        mavenLocal()
-        maven(url = "https://oss.sonatype.org/content/repositories/snapshots")
+    if (!rootProject.hasProperty("skip_snapshot_checks")) {
```
Comment (tbogdanova) [inline buildSrc/src/main/kotlin/Train.kt:58]: What is this property and where will it be used?

Comment (tbogdanova) [reply buildSrc/src/main/kotlin/Train.kt:58]: Sorry, I see we use it in our config when building libraries

Comment (tbogdanova) [reply buildSrc/src/main/kotlin/Train.kt:58]: But do I understand correctly that this parameter only makes sense when building ktor?

Comment (tbogdanova) [reply buildSrc/src/main/kotlin/Train.kt:58]: for atomicfu, coroutines, serialization in project branches I also see logic with this parameter. 
I will add it to ktor build for train.

Comment (osipxd) [reply buildSrc/src/main/kotlin/Train.kt]: This property might be useful for running train locally with different versions, but I don't think it should ever be used on CI

**Pull request merged**